### PR TITLE
Bug fix/agency update endpoints

### PR DESCRIPTION
**Pull request by kvahed**
Fixes arangodb/team-clifton-planning#38

### CODE REVIEW

File: arangod/Agency/State.cpp [hunk 1]
```diff
@@ -136,10 +136,90 @@ bool State::persist(index_t index, term_t term,
...
+    << " entry: " << entry.toJson();
+  
+  // The conventional log entry-------------------------------------------------
+  Builder log;
+  { VPackObjectBuilder b(&log);
+    log.add("_key", Value(stringify(index)));
+    log.add("term", Value(term));
+    log.add("request", entry);
+    log.add("clientId", Value(clientId));
+    log.add("timestamp", Value(timestamp())); }
```
Comment (lamai93) [inline arangod/Agency/State.cpp]: static strings :P

File: arangod/Agency/State.cpp [hunk 2]
```diff
@@ -159,15 +239,22 @@ std::vector<index_t> State::logLeaderMulti(query_t const& transactions,
...
-      THROW_ARANGO_EXCEPTION_MESSAGE(30000,
-                                     "Transaction syntax is [{<operations>}, "
-                                     "{<preconditions>}, \"clientId\"]");
+      THROW_ARANGO_EXCEPTION_MESSAGE(
+        30000,
+        "Transaction syntax is [{<operations>}, <preconditions>}, \"clientId\"]");
     }
 
-    if (applicable[j]) {
+    if (applicable[j] == 0) {
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Agency/State.cpp]: good call, mr. maier!

File: arangod/Agency/State.cpp [hunk 3]
```diff
@@ -310,10 +401,20 @@ index_t State::logFollower(query_t const& transactions) {
...
+      auto query = slice.get("query");
+      TRI_ASSERT(query.isObject());
+      TRI_ASSERT(query.length() > 0);
+
+      auto term = slice.get("term").getUInt();
+      auto clientId = slice.get("clientId").copyString();
+      auto index = slice.get("index").getUInt();
+      
+      bool reconfiguration =
+        query.keyAt(0).copyString().compare(0, RECONFIGURE.size(), RECONFIGURE) == 0;
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Agency/State.cpp]: true

File: arangod/Agency/Agent.cpp [hunk 4]
```diff
@@ -1328,19 +1328,23 @@ void Agent::persistConfiguration(term_t t) {
...
-        }}}}
+          agency->add("op", VPackValue("set"));
+          agency->add(VPackValue("new"));
+          { VPackObjectBuilder aa(agency.get());
+            agency->add("term", VPackValue(t));
+            agency->add("id", VPackValue(id()));
+            agency->add("active", _config.activeToBuilder()->slice());
+            agency->add("pool", _config.poolToBuilder()->slice());
+            agency->add("size", VPackValue(size()));
+            agency->add("timeoutMult", VPackValue(_config.timeoutMult()));
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Agency/Agent.cpp]: yes sir

File: arangod/Agency/State.cpp (hunk 1 above)
Comment (kvahed) [reply arangod/Agency/State.cpp]: this one will have to wait for a real cleanup

File: arangod/Agency/Agent.cpp [hunk 5]
```diff
@@ -1739,12 +1744,65 @@ query_t Agent::gossip(query_t const& in, bool isCallback, size_t version) {
...
+        }
+        
+        if (max_index > 0) { // We have a RAFT index. Wait for the RAFT commit.
+          auto result = waitFor(max_index);
+          if (result != Agent::raft_commit_t::OK) {
+#warning Report error
+          }
+        }
+      } else {
+#warning Report error
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Agency/Agent.cpp]: already done

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1191/

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1223/

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1299/
is all good

**Pull request merged**